shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx

```diff
@@ -0,0 +1,540 @@
+    // Calls registerObserver to listen to connection events
+    auto connObserver = std::make_shared<SampleConnectionObserver>();
+    connection->registerObserver(connObserver.get());
+    ```
+
+3. Call `connect` to connect to an Agora RTC channel:
+
+    ```C++
+    // Calls connect to connect to an Agora RTC channel
+    if (connection->connect(options.appId.c_str(), options.channelId.c_str(),
+                            options.userId.c_str())) {
+        AG_LOG(ERROR, "Failed to connect to Agora channel!");
+        return -1;
+    }
+    ```
+
+## Send media streams to the client
+
+Refer to the following steps to send media streams to the client:
+
+### Step 1: Create a media stream sender
+
+You can use the `IMediaNodeFactory` object to create various types of media stream senders:
+
+- `IAudioPcmDataSender`：Sends audio data in PCM format.
+- `IVideoFrameSender`：Sends video data in YUV format.
+- `IAudioEncodedFrameSender`：Sends encoded audio data.
+- `IVideoEncodedImageSender`：Sends encoded video data.
+
+1. Create an `IMediaNodeFactory` object:
+
+    ```cpp
+    // Creates an IMediaNodeFactory object.
+    agora::agora_refptr<agora::rtc::IMediaNodeFactory> factory = service->createMediaNodeFactory();
+    if (!factory) {
+    AG_LOG(ERROR, "Failed to create media node factory!");
+    }
+    ```
+
+2. Per your own requirements, create an `IAudioPcmDataSender` object, an `IVideoFrameSender` object, an `IAudioEncodedFrameSender` object, or an `IVideoEncodedImageSender` object for sending audio in PCM format, video in YUV format, encoded audio, or encoded video:
+
+    ```cpp
+    // Creates a sender for PCM audio
+    agora::agora_refptr<agora::rtc::IAudioPcmDataSender> audioPcmDataSender =
+        factory->createAudioPcmDataSender();
+    if (!audioPcmDataSender) {
+    AG_LOG(ERROR, "Failed to create audio data sender!");
+    return -1;
+    }
+
+    // Creates a sender for YUV video
+    agora::agora_refptr<agora::rtc::IVideoFrameSender> videoFrameSender =
+        factory->createVideoFrameSender();
+    if (!videoFrameSender) {
+    AG_LOG(ERROR, "Failed to create video frame sender!");
+    return -1;
+    }
+
+    // Creates a sender for encoded audio
+    agora::agora_refptr<agora::rtc::IAudioEncodedFrameSender> audioFrameSender =
+        factory->createAudioEncodedFrameSender();
+    if (!audioFrameSender) {
+        AG_LOG(ERROR, "Failed to create audio encoded frame sender!");
+        return -1;
+    }
+    
+    // Creates a sender for encoded video
+    agora::agora_refptr<agora::rtc::IVideoEncodedImageSender> videoEncodedFrameSender =
+        factory->createVideoEncodedImageSender();
+    if (!videoEncodedFrameSender) {
+    AG_LOG(ERROR, "Failed to create encoded video frame sender!");
+    return -1;
+    }
+    ```
+
+3. Create an `ILocalAudioTrack` object and an `ILocalVideoTrack` object, which correspond to the local audio track and local video track to be published to the Agora RTC channel:
+
+    ```c++
```

Comment:
Code highlighting looks off. Maybe something is wrong with the rendering.

Reply:
Fixed.